lib/spack/spack/cmd/license.py

```
@@ -178,14 +185,110 @@ def verify(args):
+        return []
+
+    # easy case: license looks mostly familiar
+    start = next((i for i, line in enumerate(lines) if re.match(r"#\s*Copyright", line)), -1)
+    end = next((i for i, line in enumerate(lines) if re.match(r"#\s*SPDX-", line)), -1)
+
+    # here we just replace a bad license with the fixed one
+    if start >= 0 and end >= 0:
+        # filter out weird cases and make sure we mostly know what we're fixing
+        if (
+            end < start
+            or end - start > 6
+            or not all(lines[i].startswith("#") for i in range(start, end))
+        ):
+            return []
+
+        if start < (license_lines - len(license_line_regexes)):
+            # replace license where it is
+            lines[start : end + 1] = fixed_lines
+        else:
+            # move license to beginning of file
+            del lines[start : end + 1]
+
+            start = 0
+            while any(lines[start].startswith(s) for s in ("#!", "# -*-")):
+                start += 1
+
+            lines[start:start] = fixed_lines
+
+        return lines
+
+    # no license in the file yet, so we add it
+    if start == -1 and end == -1:
+        start = 0
+        while any(lines[start].startswith(s) for s in ("#!", "# -*-")):
+            start += 1
+
+            # add an empty line if needed
+            if not re.match(r"#\s*$", lines[start]):
+                lines[start:start] = "#"
+                start += 1
+
+        lines[start:start] = fixed_lines
+        return lines
+
+    return []
+
+
+def fix(args):
+    """Fix files without proper licenses."""
+    license_errors = _find_license_errors(args)
+    if not license_errors.has_errors():
+        tty.msg("No license issues found.")
+        return
+
+    returncode = 0
+    for error_type, path in license_errors.errors:
+        lines = _fix_path(path)
+        if not lines:
+            print(f"I don't know how to fix {path}")
```

tty.die?

---

it's meant to enumerate all the files it can't fix -- do you want it to fail fast?